### ISSUE TITLE
Callbacks and proper error handling

### ISSUE THREAD

**Issue by llaakso** [bwindels/exif-parser#13]
This module seems handy but it crashes in node.js environment:

node_modules/exif-parser/lib/jpeg.js:12
                                throw new Error('Invalid JPEG section offset');
                                ^

Error: Invalid JPEG section offset
    at Object.module.exports.parseSections (/home/xu/node_modules/exif-parser/lib/jpeg.js:12:11)

In node you would avoid this by using callbacks to indicate error.

**Comment by bpatrik** [comment]
I think that shouldn't be an error after all, just a 'warning'. 
The module menages to get some data from broken images as well, but with this throw, those data are lost.

**Comment by bwindels** [comment]
Thank you for your suggestion.

As this is not an async API, adopting the node pattern of `(err, result)` makes little sense to me. The callback is used for the low level API as a visitor pattern, to avoid creating temporary arrays.

Also, it's way easier to just throw on unrecoverable errors (like an invalid JPEG section offset) than to have error handling at every layer of the callstack.

It's also easy enough to just slap a `try` around it, just like you would with `JSON.parse`.

You do have a point that it should be documented that the library will throw on invalid data, I just added this to the README.

**Comment by bwindels** [comment]
Hmm, as @bpatrik mentioned, it probably shouldn't throw once it has already found the exif section, and just stop parsing once all information has been found.

**Comment by MickL** [comment]
I agree that this module should not throw an error at all. I've used hundreds of modules but don't know any that requires a try-block somewhere.

**Comment by kwaping** [comment]
Hi @bwindels do you have a plan to address this? I like your proposal in https://github.com/bwindels/exif-parser/issues/13#issuecomment-408247330 .

**Comment by bwindels** [comment]
Don't have time to work on this in the near future, sorry. Happy to review PRs though.

**Comment by Mulperi** [comment]
I know this is 4 years old post but I just tried this library and encountered this error trying to do batch read with array that also had directory names inside.
